workflows/diagnostics/fv3net/diagnostics/prognostic_run/compute.py

```diff
@@ -101,7 +102,7 @@ def _merge_diag_computes(
             )
             continue
 
-        diag_arg = DiagArg(prog, verif, grid)
+        diag_arg = DiagArg(prog, verif, grid, gsrm=gsrm)
```

Comment:
Is it possible for the `DiagArg` class to keep its old signature (without gsrm arg) and just figure out what its horizontal_dims attribute should be based off the grid or prediction datasets provided? I see there's already a `check_if_scream_dataset` function added for convenience.

Reply:
Good point, we can actually keep the old signature and not have `gsrm`or `model` floating around at all.